src/pinnable_mapped_file.cpp

```diff
@@ -265,36 +264,39 @@ void pinnable_mapped_file::revert_to_private_mode() {
 }
 
 // returns the number of pages flushed to disk
-size_t pinnable_mapped_file::check_memory_and_flush_if_needed() {
+std::optional<std::pair<size_t, size_t>> pinnable_mapped_file::check_memory_and_flush_if_needed() {
    size_t written_pages {0};
-#if 0
    if (_non_file_mapped_mapping || _sharable || !_writable)
-      return written_pages;
+      return {};
 
    // we are in `copy_on_write` mode.
    static time_t check_time = 0;
-   constexpr int check_interval = 60; // seconds
-   constexpr size_t one_gb = 1ull << 30;
+   constexpr int check_interval = 30; // seconds
```

Comment:
I wonder if it is worth reducing the check_interval once oom_score is above some threshold? For example, check every 1s after it is above 900.

Reply:
I don't think it is needed. When the `oom` threshold is reached, all state pages are flushed (in async mode) so the system is free to pages them to disk as needed. I did test various delays and they did not make much of a difference if any.